mantle/platform/qemu.go

```diff
@@ -912,16 +943,47 @@ func setupPreboot(arch, confPath, firstbootkargs, kargs string, diskImagePath st
 			return errors.Wrapf(err, "reading bootloader config")
 		}
 		var buf strings.Builder
+
+		// BLS values required for s390x zipl
+		var linux string
+		var initrd string
+		var options string
+
 		for _, line := range strings.Split(string(origconf), "\n") {
 			if strings.HasPrefix(line, "options ") {
 				line += " " + kargs
+				if s390x_zipl_sync {
+					options = strings.TrimSpace(line+" ignition.firstboot "+firstbootkargs) + "\n"
```

Comment:
This should be `kargs` and not `firstbootkargs` here, right?
```suggestion
					options = strings.TrimSpace(line+" ignition.firstboot "+kargs) + "\n"
```

Also we should probably be reading in the contents of `/boot/ignition.firstboot` and using that instead of hardcoding `ignition.firstboot` here I think.

Reply:
no, `kargs` are part of the `line`

Reply:
ahh I see. 

Though, this code will never get reached if someone just provided `firstbootkargs` and not `kargs`.